storage-access.bs

```diff
@@ -4,27 +4,111 @@ Shortname: storage-access
+</ul>
+</div>
+
+When invoked, |document|.<dfn export method for=Document><code>requestStorageAccess()</code></dfn> must run these steps:
+
+1. Let |p| be [=a new promise=].
+1. If |document|'s [=has storage access flag=] is set, [=resolve=] |p| and abort these steps.
+1. If the |document|'s [=active sandboxing flag set=] has its [=sandboxed origin browsing context flag=] set, [=reject=] |p| and abort these steps.
+1. If |document|'s [=has storage access flag=] is unset, [=reject=] |p| and abort these steps.
+1. If |document|'s <a for=Document>browsing context</a> is a [=top-level browsing context=], [=resolve=] |p| and abort these steps.
+1. If |document|'s <a for=Document>browsing context</a> is [=same origin=] with its [=top-level browsing context=], [=resolve=] |p| and abort these steps.
+1. If the |document|'s {{Window}} object has [=transient activation=] and the |document|'s [=active sandboxing flag set=] has its [=sandbox storage access by user activation flag=] set, [=reject=] |p| and abort these steps.
+1. If |document|'s <a for=Document>browsing context</a>'s [=opener browsing context=] is not its [=top-level browsing context=], [=reject=] |p| and abort these steps.
+1. If the algorithm is invoked when the |document|'s {{Window}} object does not have [=transient activation=], [=reject=] |p| and abort these steps.
+1. <span class=XXX>Finish this algorithm.</span>
+1. [=If aborted=], return |p|.
+
+<div class=issue>
+<p>Complete this algorithm.
+<p>See its <a href="https://developer.mozilla.org/en-US/docs/Web/API/Document/requestStorageAccess">MDN page</a>, <a href="https://trac.webkit.org/browser/webkit/trunk/Source/WebCore/dom/DocumentStorageAccess.cpp#L123">WebKit implementation</a>, and <a href="https://hg.mozilla.org/mozilla-central/file/tip/dom/base/Document.cpp#l15629">Gecko implementation</a>.
+</ul>
+</div>
+
+<h3 id="cookies">Changes to {{Document/cookie}}</h3>
+
+ISSUE: Write this section.
+
+ISSUE(4): Should this API affect client-side storage other than cookies?
+
+<h3 id="sandboxing-storage-access">Sandboxing storage access</h3>
+
+A [=sandboxing flag set=] has a <dfn export>sandbox storage access by user activation flag</dfn>. This flag prevents content from requesting storage access.
+
+To the [=parse a sandboxing directive=] algorithm, add the following under step 3:
+
+<ul>
+<li>The [=sandbox storage access by user activation flag=], unless <var ignore>tokens</var> contains the <dfn export attr-value for=iframe/sandbox>allow-storage-access-by-user-activation</dfn> keyword.
+</ul>
+
+ISSUE(10): Provide mechanism for nested iframes to request storage access
+
+ISSUE(12): What about Feature Policy?
 
```

Comment:
Will the change to the HTML Living Standard cover when storage access is removed, such as when the page is navigated? Or will it only cover how storage access is checked (hasStorageAccess) and requested (requestStorageAccess)?

Reply:
I've tried to capture this in d62d30c. Let me know if that works.